### PR TITLE
cpu/kinetis: model kconfig

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This models the Kconfig modules for all boards based on kinetis. 72e7f8d30de877970694c14f9b6a5ab313bb8111 limits the CI to test only this on `mulle` for now. Some changes on drivers are also included here (mostly adding hardware features to indicate presence).

The list of enabled boards is:
- frdm-k22f
- frdm-k64f
- frdm-kl43z
- frdm-kw41z
- mulle
- openlabs-kw41z-mini
- openlabs-kw41z-mini-256kib
- pba-d-01-kw2x
- phynode-kw41z
- teensy31
- usb-kw41z

### Testing procedure
- Green CI
- Check the modelling and compare `make info-modules` list with and without `TEST_KCONFIG=1`, they should match.
- Double-check that all boards based on kinetis are here

### Issues/PRs references
Part of #16875
~~Depends on #17307~~

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
Rebased to resolve conflicts

**Comment by MrKevinWeiss** [comment]
Ugg, rebase and maybe we can rerun to check the output...

**Comment by leandrolanzieri** [comment]
#17307 is in, I rebased. Let's see what Murdock thinks

**Comment by MrKevinWeiss** [comment]
such nice error output :smile:

**Comment by leandrolanzieri** [comment]
I forgot to update the `test/driver_tmp00x` application configuration. 6f236a15d0a6dd942e9a7904018278538a2e48a7 should fix it.

**Comment by leandrolanzieri** [comment]
I added some fixups and the rest of the kinetis boards to quick-check on Murdock.

**Comment by leandrolanzieri** [comment]
It looks this is passing for all kinetis boards now

**Comment by leandrolanzieri** [comment]
I rebased to resolve conflicts.

File: boards/phynode-kw41z/Kconfig [hunk 1]
```diff
@@ -13,5 +13,8 @@ config BOARD_PHYNODE_KW41Z
     select BOARD_COMMON_KW41Z
     select HAS_PERIPH_I2C
     select HAS_PERIPH_SPI
+    select HAVE_CCS811
```
Comment (MrKevinWeiss) [inline boards/phynode-kw41z/Kconfig:16]: not a big deal but is seems there is a convention to have a newline to separate `HAS_` and `HAVE_`

**Review by MrKevinWeiss — COMMENTED** [review]
A quick look over it looks good, let's wait for murdock I guess.

File: boards/phynode-kw41z/Kconfig (hunk 1 above)
Comment (MrKevinWeiss) [reply boards/phynode-kw41z/Kconfig:16]: oops, wanted to delete this one... It is super minor.

**Comment by MrKevinWeiss** [comment]
> Double-check that all boards based on kinetis are here

It also looks like all the boards are included

**Comment by MrKevinWeiss** [comment]
Please change the .,murdock file and squash!

**Comment by leandrolanzieri** [comment]
> Please change the .,murdock file and squash!

Done!

File: boards/phynode-kw41z/Kconfig (hunk 1 above)
Comment (leandrolanzieri) [reply boards/phynode-kw41z/Kconfig:17]: Thanks, I fixed this.

**Comment by leandrolanzieri** [comment]
All green here

**Comment by fjmolinas** [comment]
Is this waiting for anything @MrKevinWeiss ?

**Review by MrKevinWeiss — APPROVED** [review]
ACK!

**Comment by leandrolanzieri** [comment]
Thanks for the review!

**Pull request merged**